### ISSUE TITLE
prefer-const is incorrectly marking arrays as const when a push is used

### ISSUE THREAD

**Issue by PhilHannent** [palantir/tslint#4343]
### Bug Report

- __TSLint version__: 5.11
- __TypeScript version__: 3.1.4
- __Running TSLint via__: VSCode

#### TypeScript code being linted

```ts
let knownVersions: number[] = []
```

with `tslint.json` configuration:

```json
{
	"defaultSeverity": "warning",
	"extends": [
		"tslint:recommended"
	],
	"jsRules": {},
	"rules": {
		"max-line-length": [
			false
		],
		"indent": [
			true,
			"tabs"
		],
		"semicolon": [
			true,
			"never"
		],
		"quotemark": [
			true,
			"single",
			"avoid-escape",
			"avoid-template"
		],
		"trailing-comma": [
			true,
			{
				"multiline": "never",
				"singleline": "never"
			}
		],
		"no-console": [
			false
		],
		"no-namespace": false,
		"max-classes-per-file": false
	},
	"rulesDirectory": []
}
```

#### Actual behavior
Whenever I have an array which is only added to using the .push() function, the Linter will convert it into a const.

#### Expected behavior
I would expect the array to stay as a let variable.

**Comment by JoshuaKGoldberg** [comment]
Hi @PhilHannent, thanks for posting this! This is a common annoyance with `const` - it doesn't refer to the object being immutable, just that the _reference_ to the object (the variable) doesn't change. This is totally fine too:

```
const box: { [i: string]: number } = {};

box.foo = 7;
```

If you want your values to be immutable, consider using the built-in `ReadonlyArray<T>` type or adopting rules from [tslint-immutable](https://www.npmjs.com/package/tslint-immutable).
